app/models/manageiq/providers/vmware/network_manager/refresh_parser.rb

```diff
@@ -0,0 +1,207 @@
+
+      new_result = {
+        :name          => network.name,
+        :ems_ref       => uid,
+        :enabled       => true,
+        :shared        => network.is_shared,
+        :type          => network_type,
+        :cloud_subnets => []
+      }
+      return uid, new_result
+    end
+
+    def parse_network_subnet(network)
+      uid = subnet_id(network)
+      new_result = {
+        :name            => subnet_name(network),
+        :ems_ref         => uid,
+        :gateway         => network.gateway,
+        :dns_nameservers => [network.dns1, network.dns2],
+        :type            => self.class.cloud_subnet_type,
+        :network_ports   => [],
+      }
+
+      # assign myself to the network
+      @data_index.fetch_path(:cloud_networks, network.id)[:cloud_subnets] << new_result
+
+      return uid, new_result
+    end
+
+    def parse_network_port(net_if)
+      uid = port_id(net_if)
+      vm_uid = net_if[:vm].id
+
+      new_result = {
+        :type       => self.class.network_port_type,
+        :name       => port_name(net_if),
+        :ems_ref    => uid,
+        :device_ref => vm_uid,
+        :device     => @ems.vms.try(:where, :ems_ref => vm_uid).try(:first),
+      }
+
+      # find network by name, since network name is all what we are given from FOG
+      # TODO: update FOG since network *name* is not unique - network *id* should be provided
+      network = @data[:cloud_networks].find { |n| n[:name] == net_if[:network] }
+      network ||= @data[:cloud_networks].find do |n|
+        n[:name] == vapp_network_name(net_if[:network], net_if[:vapp])
+      end
+
+      unless network.nil?
+        subnet = network[:cloud_subnets].first
+        cloud_subnet_network_port = {
+          :address      => net_if["ip_address"],
+          :cloud_subnet => subnet
+        }
+        new_result[:cloud_subnet_network_ports] = [cloud_subnet_network_port]
+      end
+
+      return uid, new_result
+    end
+
+    # Utility
+
+    def fake_vapp_networks
+      # TODO: update FOG so that it will provide list of vApp networks. Currently it is only possible to obtain their
+      # names therefore we need to fake at least minimal data to show the structure
+      require 'ostruct'
+      vdc_network_names = Set.new @inv[:networks].map(&:name)
+      vapp_networks = []
+      @org.vdcs.each do |vdc|
+        vdc.vapps.each do |vapp|
+          vapp.network_config.map do |net_conf|
+            name = net_conf[:networkName]
+
+            next if vdc_network_names.include? name
+
+            vapp_networks << OpenStruct.new(
+              :id        => vapp_network_id(name, vapp),
+              :name      => vapp_network_name(name, vapp),
+              :is_shared => nil,
+              :type      => "application/vnd.vmware.vcloud.vAppNetwork+xml",
+              :gateway   => nil,
+              :dns1      => nil,
+              :dns2      => nil,
+            )
```

Comment:
If the attributes are known and constant I think its better to use `Struct` vs `OpenStruct`